mods/tnt/init.lua

```diff
@@ -333,24 +333,25 @@ local function tnt_explode(pos, radius, ignore_protection, ignore_on_blast)
 	vm:update_liquids()
 
 	-- call nodeupdate for everything within 1.5x blast radius
+	for y = -radius * 1.5, radius * 1.5 do
 	for z = -radius * 1.5, radius * 1.5 do
 	for x = -radius * 1.5, radius * 1.5 do
-	for y = -radius * 1.5, radius * 1.5 do
-		local s = vector.add(pos, {x = x, y = y, z = z})
-		local r = vector.distance(pos, s)
+		local rad = {x = x, y = y, z = z}
+		local s = vector.add(pos, rad)
```

Comment:
Oh, you could also move the `vector.add` call into the conditional so that it isn't called when not necessary.  So, something like `nodeupdate_single(vector.add(pos, rad))`.
(Sorry about so many separate comments, this isn't critical anyways and it can be fixed on merging or even left as-is).